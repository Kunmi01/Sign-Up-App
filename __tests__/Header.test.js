import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ConnectedHeader, { Header } from '../src/components/Header';

describe('Header', () => {
  const initialState = {
    userData: {
      name: '',
      role: '',
      email: '',
      password: ''
    },
    privacyData: {
      trayUpdates: null,
      productEmails: null
    }
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedHeader />
        </BrowserRouter>
      </Provider>
    );
  });

  it('renders connected(SMART) component', () => {
    expect(container.find(ConnectedHeader).exists()).toBe(true);
    expect(toJson(container)).toMatchSnapshot();
  });

  it('prop match with initial store state', () => {
    const headerContainer = container.find(Header);
    expect(headerContainer.prop('state')).toEqual(initialState);
  });

  it('contains sign up navigation links', () => {
    const headerContainer = container.find(Header);

    expect(headerContainer.find({ to: '/user' })).toHaveLength(1);
    expect(headerContainer.find({ to: '/privacy' })).toHaveLength(1);
    expect(headerContainer.find({ to: '/done' })).toHaveLength(1);
  });
});
