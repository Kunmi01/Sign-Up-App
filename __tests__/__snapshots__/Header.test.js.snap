// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders connected(SMART) component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          dispatch={[Function]}
          state={
            Object {
              "privacyData": Object {
                "productEmails": null,
                "trayUpdates": null,
              },
              "userData": Object {
                "email": "",
                "name": "",
                "password": "",
                "role": "",
              },
            }
          }
        >
          <div
            className="header"
          >
            <div
              className="header__wrapper"
            >
              <NavLink
                activeClassName="active"
                className="header__link header__link--user"
                to="/user"
              >
                <Link
                  aria-current={null}
                  className="header__link header__link--user"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/user",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="header__link header__link--user"
                    href="/user"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="header__link header__link--user"
                      href="/user"
                      onClick={[Function]}
                    >
                      User
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="header__link header__link--privacy disabled"
                to="/privacy"
              >
                <Link
                  aria-current={null}
                  className="header__link header__link--privacy disabled"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/privacy",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="header__link header__link--privacy disabled"
                    href="/privacy"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="header__link header__link--privacy disabled"
                      href="/privacy"
                      onClick={[Function]}
                    >
                      Privacy
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="header__link header__link--done disabled"
                to="/done"
              >
                <Link
                  aria-current={null}
                  className="header__link header__link--done disabled"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/done",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="header__link header__link--done disabled"
                    href="/done"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="header__link header__link--done disabled"
                      href="/done"
                      onClick={[Function]}
                    >
                      Done
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </div>
          </div>
        </Header>
      </Connect(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;
